salt/states/k8s.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+'''
+Manage Kubernetes
+
+.. code-block:: yaml
+
+    kube_label_1:
+      k8s.label_present:
+        - name: mylabel
+        - value: myvalue
+        - node: myothernodename
+        - apiserver: http://mykubeapiserer:8080
+
+    kube_label_2:
+      k8s.label_absent:
+        - name: mylabel
+        - node: myothernodename
+        - apiserver: http://mykubeapiserer:8080
+
+    kube_label_3:
+      k8s.label_folder_present:
+        - name: mylabel
+        - node: myothernodename
+        - apiserver: http://mykubeapiserer:8080
+
+
+The behavior of this module is to create dashboards if they do not exist, to
+add rows if they do not exist in existing dashboards, and to update rows if
+they exist in dashboards. The module will not manage rows that are not defined,
+allowing users to manage their own custom rows.
+'''
+
+from __future__ import absolute_import
+
+import requests
+
+# TODO should you DictDiffer
+#from salt.utils.dictdiffer import DictDiffer
+
+__virtualname__ = 'k8s'
+
+
+def __virtual__():
+    """Load always.
+
+    .. todo:: Check if requests is installed
```

Comment:
`requests` should be installed since it is a dependency but yes, this would make it future-proof since it may be removed.